test/algo/compression/v2/test_auto_conv.py

```diff
@@ -0,0 +1,154 @@
+        self.fc1 = torch.nn.Linear(16 * 4 * 4, 120)
+        self.fc2 = torch.nn.Linear(120, 84)
+        self.fc3 = torch.nn.Linear(84, 10)
+        self.pool1 = torch.nn.MaxPool2d((2, 2))
+        self.pool2 = torch.nn.MaxPool2d((2, 2))
+        self.cond = torch.jit.script(CondModel())
+        self.asub = ASubModel()
+
+    def forward(self, x: torch.Tensor):
+        x = x.contiguous(memory_format=torch.channels_last)
+        x = torch._C._nn.upsample_bilinear2d(x, (28, 28), False)
+        x = torch._C._nn.upsample_nearest2d(x, (28, 28))
+        x = F.adaptive_avg_pool2d(x, (28, 28))
+
+        x = torch.exp(x)
+        x = torch.sigmoid(x)
+
+        x = torch.transpose(x, 1, 2)
+        x = torch.transpose(x, 1, 2)
+
+        x = F.avg_pool2d(x, 3, 1, padding=1)
+        x = F.max_pool2d(x, 3, 1, padding=1)
+
+        x = x.to(torch.float32)
+
+        y1 = self.pool1(F.relu(self.conv1(x)))
+        y2 = self.pool1(F.gelu(self.conv1(x)))
+
+        x = y1 + y2
+
+        x = x + 0.00001
+
+        x = x * 1.00001
+
+        y1 = self.pool2(F.silu(self.conv2(x)))
+        y2 = self.pool2(torch.tanh(self.conv2(x)))
+
+        x = y1 - y2
+
+        x = x - 0.00001
+
+        x = x / 1.00001
+
+        x = torch.permute(x, (0, 2, 3, 1))
+        x = torch.permute(x, (0, 2, 3, 1))
+        x = torch.permute(x, (0, 2, 3, 1))
+        x = torch.unsqueeze(x, dim=1)
+        x = torch.select(x, dim=1, index=0)
+        x = torch.unsqueeze(x, dim=1)
+        x = torch.mean(x, dim=1)
+        x = torch.unsqueeze(x, dim=1)
+        x = torch.sum(x, dim=1, dtype=torch.float32)
+        x = torch.unsqueeze(x, dim=1)
+        x = torch.squeeze(x, dim=1)
+        x = torch.flatten(x, 1)
+        x = x.reshape(x.shape)
+        x = x.view(-1, x.size(1))
+
+
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = F.softmax(self.fc3(x), dim=1)
+
+        y1 = x[:,0:int(x.size(1)/2)]
+        y2 = x[:,int(x.size(1)/2):x.size(1)]
+        x = torch.cat((y1, y2), dim=1)
+
+        x = x.type_as(x)
+        x = x.expand_as(x)
+        x = torch.matmul(x, x.t())
+        x = torch.cat([x, x], dim=1)
+        # x = self.cond(x)
+        x = self.asub(x)
+        x = torch.constant_pad_nd(x, (1,1,1,1), 3.14159)
+
+        return x
+
+class AutoConvTestCase(unittest.TestCase):
+    def test_level_pruner(self):
+        model = TorchModel1()
+        dummy_input = torch.rand(3, 1, 28, 28)
+        config_list = [{'op_types': ['Conv2d'], 'sparsity': 0.8}]
+        pruner = LevelPruner(model=model, config_list=config_list)
+        pruned_model, masks = pruner.compress()
+        pruner._unwrap_model()
+        sparsity_list = compute_sparsity_mask2compact(pruned_model, masks, config_list)
+        assert 0.78 < sparsity_list[0]['total_sparsity'] < 0.82
+        from nni.compression.pytorch.speedup import ModelSpeedup
+        ModelSpeedup(model, dummy_input, masks).speedup_model()
```

Comment:
could add `compute_sparsity_compact2origin(TorchModel1(), model, config_list)` to check sparsity ratio after speedup here

Reply:
ok